compiler/il/OMRBlock.cpp

```diff
@@ -1334,23 +1334,27 @@ OMR::Block::splitEdge(TR::Block *from, TR::Block *to, TR::Compilation *c, TR::Tr
       insertionExit->join(newBlock->getEntry());
       newBlock->getExit()->join(insertionEntry);
 
-      if (isExceptionEdge)
+      //OSRCatchBlock is handled differently from normal catch block
+      //the control flow is from -> exceptionEdge -> newBlock -> normalEdge -> to
+      if (isExceptionEdge && to->isOSRCatchBlock())
          {
-         if (to->isOSRCatchBlock())
-            newBlock->setHandlerInfoWithOutBCInfo(
-                  TR::Block::CanCatchOSR, 
-                  to->getInlineDepth(), 
-                  -1, 
-                  to->getOwningMethod(), 
-                  c);
-         else
-            newBlock->setHandlerInfo(
-               to->getCatchType(),
-               to->getInlineDepth(),
-               to->getHandlerIndex(),
-               to->getOwningMethod(),
+         newBlock->setHandlerInfoWithOutBCInfo(
```

Comment:
Thanks a lot for pointing this out. Modified the code to make the catch block match BCInfo of induction point.